pkg/api/v2/reporter/reporter.go

```diff
@@ -314,6 +343,10 @@ func (r *reporter) attemptUpdateStatus(ctx context.Context, client ReporterResou
 }
 
 func (r *reporter) StatusFromReport(report Report, subresourceStatuses map[string]*core.Status) *core.Status {
+	var messages []string
+	if len(report.Messages) != 0 {
+		messages = report.Messages
+	}
```

Comment:
how is this different than just always passing report.Messages? https://go.dev/play/p/4k0jGHToZC4

Reply:
This is just a check so that messages will always be nil rather than an an empty array. Figured this might be good futureproofing